StudyHub/StudyHub.BLL/Services/StudentAnswerService.cs

```diff
@@ -0,0 +1,121 @@
+﻿using StudyHub.BLL.Services.Interfaces;
+using StudyHub.Common.DTO.User.Student;
+using StudyHub.DAL.Repositories.Interfaces;
+using StudyHub.Entities;
+using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using StudyHub.BLL.Extensions;
+using StudyHub.Common.Exceptions;
+
+namespace StudyHub.BLL.Services;
+
+public class StudentAnswerService : IStudentAnswerService
+{
+    private readonly IRepository<StudentAnswer> _studentAnswerRepository;
+    private readonly IRepository<Assignment> _assignmentRepository;
+    private readonly IRepository<TaskOption> _taskOptionRepository;
+    private readonly IRepository<StartingTimeRecord> _startingTimeRepository;
+    private readonly IMapper _mapper;
+
+    public StudentAnswerService(
+        IRepository<StudentAnswer> studentAnswerRepository,
+        IRepository<Assignment> assignmentRepository,
+        IRepository<TaskOption> taskOptionRepository,
+        IRepository<StartingTimeRecord> startingTimeRepository,
+        IMapper mapper)
+    {
+        _studentAnswerRepository = studentAnswerRepository;
+        _assignmentRepository = assignmentRepository;
+        _taskOptionRepository = taskOptionRepository;
+        _startingTimeRepository = startingTimeRepository;
+        _mapper = mapper;
+    }
+
+    public async Task<bool> UpsertStudentAnswerAsync(Guid studentId, StudentAnswerDTO dto)
+    {
+        var startTime = await _startingTimeRepository.FirstOrDefaultAsync(x => x.StudentId == studentId)
+            ?? throw new NotFoundException("Starting time not found");
+
+        var duraction = await _assignmentRepository
+            .Where(x => x.Id == dto.AssignmentId)
+            .Select(x => x.Duration)
+            .FirstOrDefaultAsync();
+
+        var isTimeOver = DateTime.Now - startTime.StartTime < duraction;
```

Comment:
just my opinion that may it dirrectly in if might be( if u dont thing so - ok)

Reply:
?

Reply:
check comment again